internal/api/sms_provider/sms_provider.go

```diff
@@ -46,12 +53,13 @@ func GetSmsProvider(config conf.GlobalConfiguration) (SmsProvider, error) {
 	}
 }
 
-func IsValidMessageChannel(channel string, smsProvider string) bool {
+func IsValidMessageChannel(channel string, config *conf.GlobalConfiguration) bool {
+	smsProvider := config.Sms.Provider
 	switch channel {
 	case SMSProvider:
 		return true
 	case WhatsappProvider:
-		return smsProvider == "twilio" || smsProvider == "twilio_verify"
+		return smsProvider == "twilio" || smsProvider == "twilio_verify" || config.Hook.SendSMS.Enabled
```

Comment:
whatsapp should still be allowed as a valid message channel as long as the SMS hook is enabled

Reply:
Initial consideration was to ignore channel specification as [we don't relay it to the hook input](https://github.com/supabase/auth/blob/ac14e82b33545466184da99e99b9d3fe5f3876d9/internal/hooks/auth_hooks.go#L37) and have limited control over whether it is respected (e.g. developer can call the Hook with `channel`: `whatsapp` but it would only be respected if their hook implementation also sends on the WhatsApp channel.

We can expose the channe in `SMSInput` and document that it's only passed down to the hook in a separate PR though

Reply:
ah i see, ok then it's fair to return an error message if the channel specified is whatsapp i guess - will revert to retain original behaviour

Reply:
on hindsight, i've decided to keep the change since the channel specification should be ignored rather than returning the invalid channel error

i don't think we need to pass down the channel in `SMSInput` for now and indicate in the docs that the channel is not respected if a hook is being used